alchemist-physical-agents/src/main/kotlin/it/unibo/alchemist/model/implementations/properties/PhysicalPedestrian2D.kt

```diff
@@ -25,15 +26,34 @@ import org.apache.commons.math3.random.RandomGenerator
  * Base implementation of a pedestrian's capability to experience physical interactions in a 2D space.
  */
 class PhysicalPedestrian2D<T>(
-    randomGenerator: RandomGenerator,
+    private val randomGenerator: RandomGenerator,
     /**
      * The environment in which the node is moving.
      */
-    val environment: Physics2DEnvironment<T>,
+    override val environment: Physics2DEnvironment<T>,
     node: Node<T>,
-) : PhysicalPedestrian<T, Euclidean2DPosition, Euclidean2DTransformation, Euclidean2DShapeFactory> by
-PhysicalPedestrian(randomGenerator, environment, node),
-    PhysicalPedestrian2DProperty<T> {
+) : PhysicalPedestrian<T, Euclidean2DPosition, Euclidean2DTransformation, Euclidean2DShapeFactory>(
+    randomGenerator,
+    environment,
+    node,
+),
+    PhysicalPedestrian2D<T> {
+
     override val comfortArea: Euclidean2DShape get() = environment
-        .shapeFactory.circle(node.asProperty<T, AreaProperty<T>>().shape.radius + comfortRay)
+        .shapeFactory
+        .circle(node.asProperty<T, AreaProperty<T>>().shape.radius + comfortRay)
+        .transformed { origin(environment.getPosition(node)) }
+
+    override fun physicalForces(
+        environment: PhysicsEnvironment<T, Euclidean2DPosition, Euclidean2DTransformation, Euclidean2DShapeFactory>,
+    ) = environment.getNodesWithin(comfortArea)
+        .minusElement(node)
+        .filter { it.asPropertyOrNull<T, AreaProperty<T>>() != null }
```

Comment:
```suggestion
        .filterNotNull { it.asPropertyOrNull<T, AreaProperty<T>> }
```

Reply:
`filterNotNull` takes no arguments and filters non-null elements in the pipeline